### ISSUE TITLE
An UserWarning

### ISSUE THREAD

**Issue by zgplvyou** [layumi/Person_reID_baseline_pytorch#33]
UserWarning: invalid index of a 0-dim tensor. This will be an error in PyTorch 0.5. Use tensor.item() to convert a 0-dim tensor to a Python number  
  train_loss += loss.data[0]  

so transforming the  'train_loss += loss.data[0]  '  into  'train_loss+=loss.item() '  in the 'train.py:200' can fix the warning on the new pytorch version.

**Comment by layumi** [comment]
Thank you @zgplvyou  for reporting it. 
I have added it to the README.

**Comment by layumi** [comment]
I have added the code to automatically select pytorch version.

**Comment by biswajitcsecu** [comment]
/home/kerax/usys/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:31: UserWarning: invalid index of a 0-dim tensor. This will be an error in PyTorch 0.5. Use tensor.item() to convert a 0-dim tensor to a Python number
/home/kerax/usys/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:32: UserWarning: invalid index of a 0-dim tensor. This will be an error in PyTorch 0.5. Use tensor.item() to convert a 0-dim tensor to a Python number

Iter: 0, D: 1.351, G:0.6499

**Comment by layumi** [comment]
Hi @biswajitcsecu 
You may change something in your code.
For example,

```python
version =  torch.__version__

if int(version[2]) > 3: # for the new version like 0.4.0 and 0.5.0
    running_loss += loss.item()
else :  # for the old version like 0.3.0 and 0.3.1
    running_loss += loss.data[0]
```

**Comment by biswajitcsecu** [comment]
thank you
